test/test_hyper.py

```diff
@@ -1128,6 +1129,25 @@ def test_adapter_accept_client_certificate(self):
             cert=CLIENT_PEM_FILE)
         assert conn1 is conn2
 
+    def test_adapter_close(self):
+        """
+        Tests HTTP20Adapter properly closes connections
+        """
+        s = requests.Session()
+        a = HTTP20Adapter()
+        s.mount('https://http2bin.org', a)
+        s.get('https://http2bin.org/')
+        s.close()
+
+    def test_adapter_close_context_manager(self):
+        """
+        Tests HTTP20Adapter properly closes connections via context manager
+        """
+        with requests.Session() as s:
+            a = HTTP20Adapter()
+            s.mount('https://http2bin.org', a)
+            s.get('https://http2bin.org/')
```

Comment:
I've changed the test to just mount the adapter without making a request. It raises the same error.